addons/hr_expense/models/hr_expense.py

```diff
@@ -454,6 +457,14 @@ def write(self, vals):
                     self.sheet_id.write({'employee_id': vals['employee_id']})
                 elif len(employees) > 1:
                     self.sheet_id = False
+        if sheet_ids or self.sheet_id:
+            (sheet_ids | self.sheet_id)._sync_attachments_with_expenses()
```

Comment:
sheet_ids may contain multiple sheets (since self can contain multiple expenses, each associated with different sheets).

I will correct the case for no-op case (the call does nothing, but I guess it's call can be avoided)